x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -897,7 +910,15 @@ void runComputeOnRemoteCluster(
             exchangeSink.addCompletionListener(computeListener.acquireAvoid());
             runCompute(
                 parentTask,
-                new ComputeContext(localSessionId, clusterAlias, List.of(), configuration, exchangeSource, exchangeSink),
+                new ComputeContext(
+                    localSessionId,
+                    clusterAlias,
+                    List.of(),
+                    configuration,
+                    new FoldContext(configuration.pragmas().foldLimit().getBytes()),
```

Comment:
Maybe we should have a helper method for this `new FoldContext(configuration.pragmas().foldLimit().getBytes())` to enforce consistency/reduce duplication?

Reply:
Sure.